src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -391,8 +391,17 @@ def safe_to_delete(self, svc_ids):
 
     @RESTController.Resource('GET')
     def devices(self, svc_id):
+        devices: list = CephService.send_command(
+            'mon', 'device ls-by-daemon', who='osd.{}'.format(svc_id))
+        modules = CephService.send_command('mon', 'mgr module ls')
+
+        life_expectancy_enabled = any(
+            item.startswith('diskprediction_') for item in modules['enabled_modules'])
+        for device in devices:
+            device['life_expectancy_enabled'] = life_expectancy_enabled
```

Comment:
We should think of a nicer way to deal with cross-module deps. @pereman2 as you're familiar with the ceph-mgr code, isn't there any way to detect whether another module is enabled/running? Or even better, declare that a module depends on others (new global var `MODULE_DEPENDS` or something like that?) and automatically enable/load them. What do you think?

Reply:
It should be possible ernesto, let me take a look.

Reply:
@pereman2 Any news here how to get a list of enabled mgr modules without running a `send_command`?

Reply:
@torchiaf I think `mgr.get('mgr_map')` is the way to go. Have a look at https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/controllers/mgr_modules.py#L47 how to get the required information.

Reply:
There is, MGRMap has all these info and you can get it with `mgr.get('mgr_map')`. I'll paste a snippet from controllers/mgr_modules.py where this is done.
```
        result = []
        mgr_map = mgr.get('mgr_map')
        always_on_modules = mgr_map['always_on_modules'].get(mgr.release_name, [])
        for module_config in mgr_map['available_modules']:
            module_name = module_config['name']
            if module_name not in self.ignore_modules:
                always_on = module_name in always_on_modules
                enabled = module_name in mgr_map['modules'] or always_on
                result.append({
                    'name': module_name,
                    'enabled': enabled,
                    'always_on': always_on,
                    'options': self._convert_module_options(
                        module_config['module_options'])
                })
```

Reply:
@votdev @pereman2 available modules are now fetched by `mgr.get('mgr_map')`.